### ISSUE TITLE
Policy Enforcer consumes body if a Claim extracts value from Request body and results in an error in the Java application

### ISSUE THREAD

**Issue by janstrohschein** [keycloak/keycloak#14074]
### Describe the bug

In our Spring Boot application we try to retrieve a value from the request body to evaluate it in our Javascript Policy. 
We are able to extract the value from the request body as explained in the [documentation](https://github.com/keycloak/keycloak-documentation/blob/main/authorization_services/topics/enforcer-claim-information-point.adoc) via: 
`"claim-from-json-body-object": "{request.body['/a/b/c']}"`. 
After Keycloak evaluation succeeds we get a 400 Bad Request error in Spring Boot as the request body is null. The behavior was also described on [Stack overflow](https://stackoverflow.com/questions/70108663/request-body-removed-after-keycloaks-policy-enforcer-evaluation) a few months ago. 

### Version

18

### Expected behavior

The Keycloak Policy Enforcer extracts the desired value from the request body, the evaluation succeeds and afterwards the body is still available for further processing to the Spring Boot application. 

### Actual behavior

The Policy Enforcer retrieves the request body and consumes it. Afterwards the body is no longer available in Spring Boot and the result is a 400 Bad Request error.

### How to Reproduce?

1. add configuration to a Spring Boot application to retrieve values from a request body, e.g.: 
`claimInformationPointConfig.claims[user.id]={request.body['/userId']}`
2. create a Javascript Policy that retrieves the value
3. create an Endpoint in the Java Application that is secured by the Policy

### Anything else?

The behavior was also described on [Stack overflow](https://stackoverflow.com/questions/70108663/request-body-removed-after-keycloaks-policy-enforcer-evaluation) a few months ago.

**Comment by pedroigor** [comment]
@janstrohschein Thanks. I'm aware of this issue when using the PEP on top of some stacks like SB.

We have this issue fixed for Undertow/Wildfly adapters, but we never managed to fix it for other stacks.

Would you be able to send a patch? I'm asking because otherwise, we won't work on it anytime soon.

**Comment by pedroigor** [comment]
Thanks for the feedback. We are discussing authorization services and adapter deprecation here https://github.com/keycloak/keycloak/discussions/11681.

**Comment by janstrohschein** [comment]
Hey Pedro, I have been looking into it with a colleague and we found a solution to make the body readable multiple times with the valves/filters. However, as a new sprint started we did not have the time yet to integrate it into the Keycloak Adapter or write any tests. I will follow the discussion around the adapter deprecation though. Thank you for your help!

**Comment by robinhh** [comment]
>  we found a solution to make the body readable multiple times with all the subsequent valves/filters
Would you mind to share your solution? Or did you already provide a patch?

**Comment by janstrohschein** [comment]
Hey Robin (@robinhh), I did not provide a patch, as I understood that the adapter may be deprecated soon.
I attached a file archive for you with the required files. The solution for us was to target the valves and buffer the input stream as working with the filters was already too late and the request body turned out unusable. 


[keycloak_valve_workaround.zip](https://github.com/keycloak/keycloak/files/9808615/keycloak_valve_workaround.zip)
